packages/beacon-node/src/api/impl/beacon/state/utils.ts

```diff
@@ -83,6 +83,60 @@ export async function getStateResponseWithRegen(
   return res;
 }
 
+export type GeneralValidatorStatus = routes.beacon.ValidatorStatus | "active" | "pending" | "exited" | "withdrawal";
```

Comment:
On 2nd thought it makes more sense to just declare the top-level statuses here

```suggestion
export type GeneralValidatorStatus = "active" | "pending" | "exited" | "withdrawal";
```